projects/packages/publicize/src/social-image-generator/class-setup.php

```diff
@@ -88,17 +88,12 @@ public function set_meta( $post_id, $post ) {
 			return;
 		}
 
-		$post_settings = new Post_Settings( $post_id );
-		$settings      = $post_settings->get_settings();
+		$settings = new Post_Settings( $post_id );
 
-		// If SIG has explicitly been disabled for this post, we don't need to do anything else.
-		if ( isset( $settings['enabled'] ) && $settings['enabled'] === false ) {
+		if ( ! $settings->is_enabled() ) {
 			return;
 		}
 
-		// If we're here, it's safe to assume SIG should be enabled.
-		$post_settings->update_setting( 'enabled', true );
```

Comment:
Since we now always have the meta present if SIG is enabled, we can simplify our check and remove the meta update in this hook.